File: docs/snippets/all/archetypes/entity_path.py
import dalaran as dl

dl.init("dalaran_example_entity_path", spawn=True)

dl.log(
    r"world/42/escaped\ string\!",
    dl.TextDocument("This entity path was escaped manually"),
)
dl.log(
    ["world", 42, "unescaped string!"],
    dl.TextDocument(
        "This entity path was provided as a list of unescaped strings"
    ),
)

assert dl.escape_entity_path_part("my string!") == r"my\ string\!"
assert (
    dl.new_entity_path(["world", 42, "my string!"]) == r"/world/42/my\ string\!"
)

File: docs/snippets/all/archetypes/entity_path.cpp
// Log a `TextDocument`

#include <dalaran.hpp>

int main(int argc, char* argv[]) {
    const auto rec = dalaran::RecordingStream("dalaran_example_entity_path");
    rec.spawn().exit_on_failure();

    rec.log(
        R"(world/42/escaped\ string\!)",
        dalaran::TextDocument("This entity path was escaped manually")
    );
    rec.log(
        dalaran::new_entity_path(
            {"world", std::to_string(42), "unescaped string!"}
        ),
        dalaran::TextDocument(
            "This entity path was provided as a list of unescaped strings"
        )
    );

    assert(dalaran::escape_entity_path_part("my string!") == R"(my\ string\!)");
    assert(
        dalaran::new_entity_path({"world", "42", "my string!"}) ==
        R"(/world/42/my\ string\!)"
    );
}
